transport/transport.go

```diff
@@ -506,3 +507,74 @@ func wait(ctx context.Context, closing <-chan struct{}, proceed <-chan int) (int
 		return i, nil
 	}
 }
+
+const (
+	spaceByte   = ' '
+	tildaByte   = '~'
+	percentByte = '%'
+)
+
+// grpcMessageEncode encodes the grpc-message field in the same
+// manner as https://github.com/grpc/grpc-java/pull/1517.
+func grpcMessageEncode(msg string) string {
+	if msg == "" {
+		return ""
+	}
+	lenMsg := len(msg)
+	for i := 0; i < lenMsg; i++ {
+		c := msg[i]
+		if !(c >= spaceByte && c < tildaByte && c != percentByte) {
+			return grpcMessageEncodeUnchecked(msg)
+		}
+	}
+	return msg
+}
+
+func grpcMessageEncodeUnchecked(msg string) string {
+	var buf bytes.Buffer
+	lenMsg := len(msg)
+	for i := 0; i < lenMsg; i++ {
+		c := msg[i]
+		if c >= spaceByte && c < tildaByte && c != percentByte {
+			_ = buf.WriteByte(c)
```

Comment:
What about removing the underscore and adding a comment about why the error is not checked?
Seems to me the underscore is unnecessary here..